example/heatEquation2D/src/helpers.hpp

```diff
@@ -0,0 +1,19 @@
+/* Copyright 2020 Tapish Narwal
+ * SPDX-License-Identifier: ISC
+ */
+
+#pragma once
+
+#include <alpaka/alpaka.hpp>
+
+template<typename T, typename TDim, typename TIdx>
+ALPAKA_FN_ACC T* getElementPtr(T* ptr, alpaka::Vec<TDim, TIdx> idx, alpaka::Vec<TDim, TIdx> pitch)
```

Comment:
```suggestion
template<typename T>
ALPAKA_FN_ACC T* getElementPtr(T* ptr, auto idx, auto pitch)
```
Which C++ are we on? Can we do this already?

Reply:
But it is not the same: the latter does not restrict the type to be a `Vec`, nor does it require the two types to be the same.

Reply:
Yes, of course. But it is more pleasant for the eye and we know the precise context, this will be called from. I would argue that a beginner not familiar with alpaka cares less about repeating the type information twice (thrice even if you count the `TDim`, `TIdx` template arguments) and more about handling a manageable amount of information in their head. A double reinterpret cast wrapping a manual index calculation in the function body seems to me like enough to digest for a beginner.

Reply:
Still using C++17, so no we can't do this anyway